src/prometheus/definition.go

```diff
@@ -701,3 +724,41 @@ func getRawEntityID(parentGroupLabel, groupLabel, entityID string, groups defini
 	}
 	return rawEntityID, nil
 }
+
+func WithLabelValueInMetricNameAndWithLabelValueMappedToMetricValue(rawMetricName, label, format, valueLabel string, valueMap map[string]int) definition.FetchFunc {
+	return func(groupLabel, entityID string, rawGroups definition.RawGroups) (definition.FetchedValue, error) {
+		fetchedValues := definition.FetchedValues{}
+		metricsRaw, err := definition.FromRaw(rawMetricName)(groupLabel, entityID, rawGroups)
+		if err != nil {
+			return nil, fmt.Errorf("fetching metric %q for entity %q in group %q: %w", rawMetricName, entityID, groupLabel, err)
+		}
+
+		metrics, ok := metricsRaw.([]Metric)
+		if !ok {
+			return nil, fmt.Errorf("incompatible metric type for %q. Expected: []Metric. Got: %T", rawMetricName, metricsRaw)
+		}
+
+		for _, metric := range metrics {
+			nameLabelValue, ok := metric.Labels[label]
+			if !ok {
+				return nil, fmt.Errorf("label %q for metric name not found in metric %v", label, metric)
```

Comment:
Would it be better to error here, or just `continue` in case a particular metric does not contain the label?

Reply:
It would be recommended to log this error and I don't have access to the logger here, so I'm not sure. I mean, I don't see a best-effort way forward here at the moment and I wouldn't swallow the errors.

Reply:
Fucking non-global logger... Let's error out then,